test/core/promise/sleep_test.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/lib/promise/sleep.h"
+
+#include <atomic>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "absl/synchronization/notification.h"
+
+#include <grpc/grpc.h>
+
+#include "src/core/lib/promise/race.h"
+#include "src/core/lib/promise/seq.h"
+#include "test/core/promise/test_wakeup_schedulers.h"
+
+namespace grpc_core {
+namespace {
+
+TEST(Sleep, Zzzz) {
```

Comment:
Consider using a fixture to share setup and teardown logic. Simple example: https://github.com/grpc/grpc/blob/63398540b65c775de36f1e7e370c1800aca6637d/test/cpp/server/server_builder_test.cc#L47-L53

Reply:
Moved to main instead for brevity.